app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt

```diff
@@ -299,6 +311,7 @@ class TheActivityControllerTest {
 
     //then
     verify(activity, never()).showAccessDeniedScreen(fullName)
+    verifyNoMoreInteractions(activity)
     verify(patientRepository, never()).clearPatientData()
```

Comment:
Should we also add verify no interactions with repository?

Reply:
I am only adding it for outgoing changes caused by the controller since it's a part of the public API. What the controller uses internally is an implementation detail and it makes the tests and production code covariant so I intentionally don't add these assertions for these kind of dependencies.